ml-agents/mlagents/trainers/policy/tf_policy.py

```diff
@@ -392,18 +399,52 @@ def get_update_vars(self):
         """
         return list(self.update_dict.keys())
 
-    def save_model(self, steps):
+    def checkpoint(self, model_reward: Optional[float] = None) -> None:
         """
-        Saves the model
-        :param steps: The number of steps the model was trained for
-        :return:
+        Writes an intermediate checkpoint model to memory
+        model_reward: Mean reward of the reward buffer at the time of saving
         """
+        current_step = self.get_current_step()
         with self.graph.as_default():
-            last_checkpoint = os.path.join(self.model_path, f"model-{steps}.ckpt")
-            self.saver.save(self.sess, last_checkpoint)
+            last_checkpoint = os.path.join(
+                self.model_path, f"model-{current_step}.ckpt"
+            )
+            if self.saver:
+                self.saver.save(self.sess, last_checkpoint)
             tf.train.write_graph(
                 self.graph, self.model_path, "raw_graph_def.pb", as_text=False
             )
+        brain_name = self.behavior_id.brain_name
+        checkpoint_path = f"{brain_name}-{current_step}"
+        settings = SerializationSettings(self.model_path, brain_name, checkpoint_path)
+        export_policy_model(settings, self.graph, self.sess)
+        # Store steps and file_path
+        new_checkpoint = Checkpoint(
+            int(current_step),
+            os.path.join(self.model_path, f"{settings.checkpoint_path}.nn"),
+            model_reward,
+            time.time(),
+        )
+        # Record checkpoint information
+        CheckpointManager.track_checkpoint_info(
+            brain_name, new_checkpoint, self.keep_checkpoints
+        )
+
+    def save(self, model_reward: Optional[float] = None) -> None:
```

Comment:
I know why it bothers me to call policy.checkpoint() right before policy.save(). This is two method that do more or less the same work. I think we should be able to rename this save method to `export_last_checkpoint` to make it clearer what it does.